code/modules/admin/topic.dm

```diff
@@ -400,6 +400,16 @@
 		var/target = href_list["showmessageckeylinkless"]
 		browse_messages(target_ckey = target, linkless = 1)
 
+	else if(href_list["messageread"])
+		var/datum/db_query/query_message_read = SSdbcore.NewQuery(
+			"UPDATE [format_table_name("messages")] SET type = 'message sent' WHERE id = :id",
+			list("id" = href_list["message_id"])
```

Comment:
You can't trust any input from the client.

You need to verify that this message_id is both:

A. An integer
B. An ID that the player actually has

Reply:
How do I do that? Are there any examples in code I can look at?

Reply:
Good `ui_act` procs will check.

To satisify A, check `isnum(href_list["message_id"])`, then `rond` it. That way `2.1` turns into `2`, and so it's not as big a deal.

To satisfy B, you could perform a SQL query or similar to check if they have the ID, but I think the more clever thing to do here is to add in an extra condition to the `WHERE` to verify its for the same ckey.